reactivesocket-cpp/src/Frame.cpp

```diff
@@ -114,34 +117,83 @@ std::ostream& operator<<(std::ostream& os, const FrameHeader& header) {
   std::bitset<16> flags(header.flags_);
   return os << header.type_ << "[" << flags << ", " << header.streamId_ << "]";
 }
+
+std::unique_ptr<folly::IOBuf> FrameMetadata::serialize() {
```

Comment:
I have a concern about allocating small buffers and chaining, rather than allocating the correct size buffer at the start (after computing size) and then writing into that.  I'm interested to hear other viewpoints.